src/common/url/check-url.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable no-throw-literal */
+import { warn } from '../util/console'
+import { isIE } from '../constants/runtime'
+
+/**
+ * Checks input string to see if it's acceptable as agent's public path for loading source code.
+ * @param {string} string - supposed url string
+ * @param {boolean} [allowUnsecure] - when scheme isn't included in url string, this allows http: to be prepended if true
+ * @returns Absolute url if 'string' is accepted. Empty string if it's denied.
+ */
+export function validateServerUrl (string, allowUnsecure = false, localTest = false) {
```

Comment:
We need a better way of altering this code for local testing rather than a parameter. We need a way to allow insecure for our testing purposes but remove support for it in our production builds.